### DISCUSSION TITLE
Cross device email verification

### DISCUSSION THREAD

**Discussion by simonsisay** [supabase/supabase#30219] [Questions]
I'm using Supabase with credentials-based auth and facing an edge case with cross-device email verification. Here’s the scenario:

1. A user signs up and is redirected to a verify-email route on their laptop.
2. They check their email on another device (e.g., their phone) and click the "confirm email" button.
3. Returning to their laptop, they’re still on the verify-email page, as the session doesn't reflect the verified status.

Normally, I would poll every few seconds to check if email_verified is true to automatically redirect them to the dashboard once verified. However, since there’s no active session on the verify-email page, Supabase returns null on any query for the auth user table, so I can’t check the verification status.

Is there a workaround or any suggested approach to handle this flow with Supabase? Any advice or insights would be greatly appreciated.

**Comment by Heet-Bhalodiya** [comment]
Having the same doubt, can anyone help me with that?

**Reply by RohinJ444** [reply]
Same here

**Reply by alexeytrubin** [reply]
same
